api/v1alpha1/gateway_types.go

```diff
@@ -54,78 +57,184 @@ type GatewayList struct {
 type GatewaySpec struct {
 	// Class used for this Gateway. This is the name of a GatewayClass resource.
 	Class string `json:"class" protobuf:"bytes,1,opt,name=class"`
-	// Listeners associated with this Gateway. Listeners define what addresses,
-	// ports, protocols are bound on this Gateway.
+
+	// Listeners associated with this Gateway. Listeners define logical
+	// endpoints that are bound on this Gateway. If no Listeners are
+	// provided, the GatewayClass may associate the Gateway with
+	// an address but the Gateway will not be able to accept any
```

Comment:
Why not be able to accept ? If you don't specify any listener it could default to HTTP/HTTPS.

Reply:
If there are no Listeners, there's no way to know where to send the traffic.

Reply:
If there are no listeners, you don't even know which port to listen on to accept any traffic in the first place.

Reply:
I can resolve this by making Listeners required?

Reply:
Yes -- that's probably easiest rather than trying to figure out what empty listeners mean (for now)